pyro/contrib/named.py

```diff
@@ -101,13 +163,36 @@ class List(list):
     def __init__(self, name=None):
         self._name = name
 
+    def __repr__(self):
+        return "List ( {} )".format(self._expand())
+
     def _set_name(self, name):
         if self:
             raise RuntimeError("Cannot name a named.List after data has been added")
         if self._name is not None:
             raise RuntimeError("Cannot rename named.List: {}".format(self._name))
         self._name = name
 
+    def _expand(self):
+        ls = []
```

Comment:
Could be simplified by defining a top level helper
```py
def is_named(thing):
    return isinstance(thing, (Object, List, Dict))
```
and then this function becomes
```py
return [l._expand() if is_named(l) else l for l in self]
```